library/lib/application.cpp

```diff
@@ -451,9 +511,18 @@ void Application::navigate(FocusDirection direction)
 {
     View* currentFocus = Application::currentFocus;
 
-    // Do nothing if there is no current focus or if it doesn't have a parent
+    // We should allow the user to regain focus to a default focus view,
+    // in the most recent view in the stack, in a case where the user is
+    // unable to navigate through normal means, because currentFocus is nullptr.
+    // It will happen with mouse and touch controls, it has happened to me.
+    if(!currentFocus){
+        Application::giveFocus(viewStack.back()->getDefaultFocus());
```

Comment:
That will cause issues for modal dialogs, with that the user will be able to focus what's under the dialog, that's not a good idea

Reply:
I've tested it with the Dialog class and that doesn't seem to be a concern. I think the reason for it is due to the way modals are processed on the view stack.
```
// dialog.cpp
void Dialog::open()
{ // Line 58
    ...
    Application::pushView(this); // The modal dialog is pushed to the top of the stack. It is now the most recent view on the stack.
    ...
}
```
The modifications I did only touch the modal view because it is the most recent view on the stack, and it selects whatever button is the default focus within that view. With mouse and touch, the same thing happens. A touch or click queries the most recent view on the stack so the same principle applies.

Reply:
Did you try a dialog without any button though? Because that's a thing

Reply:
No, but I will do that in one minute. Please hold.

Reply:
Ok I just did. It was not possible to click outside the modal to make it go away.

But since the modal dialogs don't have any default focus I could not find a way to make the dialog disappear. I don't think it is possible with keyboard and mouse to escape a modal dialog if it does not have buttons.

Reply:
> But since the modal dialogs don't have any default focus I could not find a way to make the dialog disappear. I don't think it is possible with keyboard and mouse to escape a modal dialog if it does not have buttons.

This is actually intended, to create modal loading dialogs for instance.

Reply:
Ok, that makes sense.